### PR TITLE
Password Strength Checker

### LINKED ISSUE #449: Checking Strength of given password In Java

### PR DESCRIPTION
**Pull request by siddharthtankariya**
This pull request fixes  #449

### CODE REVIEW

File: passwd_strength/PasswordStrength.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+public class PasswordStrength {
```
Comment (t2013anurag) [inline passwd_strength/PasswordStrength.java]: move the file inside password_generator

File: passwd_strength/PasswordStrength.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                }
+
+                else if (Character.isDigit(x)) {
+
+                    hasDigit = true;
+                }
+
+                // no need to check further, break the loop
+                if(hasLetter && hasDigit && hasUpperCase && hasLowerCase){
+
```
Comment (t2013anurag) [inline passwd_strength/PasswordStrength.java]: remove extra new line

File: passwd_strength/PasswordStrength.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+					}else if(Character.isLowerCase(x)){
+						
+						hasLowerCase = true;
+					}
+					
+                    hasLetter = true;
+                }
+
+                else if (Character.isDigit(x)) {
+
```
Comment (t2013anurag) [inline passwd_strength/PasswordStrength.java]: remove extra new line

File: passwd_strength/PasswordStrength.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                    hasDigit = true;
+                }
+
+                // no need to check further, break the loop
+                if(hasLetter && hasDigit && hasUpperCase && hasLowerCase){
+
+                    break;
+                }
+
```
Comment (t2013anurag) [inline passwd_strength/PasswordStrength.java]: remove extra new line

File: passwd_strength/PasswordStrength.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                if (Character.isLetter(x)) {
+					if(Character.isUpperCase(x)){
+						
+						hasUpperCase = true;
+						
+					}else if(Character.isLowerCase(x)){
+						
+						hasLowerCase = true;
+					}
+
```
Comment (t2013anurag) [inline passwd_strength/PasswordStrength.java]: remove extra new line

File: passwd_strength/PasswordStrength.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        if (password.length() >= 8) {
+            for (int i = 0; i < password.length(); i++) {
+                char x = password.charAt(i);
+                if (Character.isLetter(x)) {
+					if(Character.isUpperCase(x)){
+						
+						hasUpperCase = true;
+						
+					}else if(Character.isLowerCase(x)){
+
```
Comment (t2013anurag) [inline passwd_strength/PasswordStrength.java]: remove extra new line

File: passwd_strength/PasswordStrength.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        boolean hasDigit = false;
+		boolean hasUpperCase = false;
+		boolean hasLowerCase = false;
+		
+        if (password.length() >= 8) {
+            for (int i = 0; i < password.length(); i++) {
+                char x = password.charAt(i);
+                if (Character.isLetter(x)) {
+					if(Character.isUpperCase(x)){
+
```
Comment (t2013anurag) [inline passwd_strength/PasswordStrength.java]: remove extra new line

File: passwd_strength/PasswordStrength.java (hunk 4 above)
Comment (siddharthtankariya) [reply passwd_strength/PasswordStrength.java]: Done For all Lines

**Comment by t2013anurag** [comment]
@siddharthtankariya - can you move the file inside the existing password_generator directory

File: passwd_strength/PasswordStrength.java (hunk 1 above)
Comment (siddharthtankariya) [reply passwd_strength/PasswordStrength.java]: Done

**Comment by siddharthtankariya** [comment]
Moved the file.

File: passwd_strength/PasswordStrength.java (hunk 7 above)
Comment (siddharthtankariya) [reply passwd_strength/PasswordStrength.java]: Done

File: passwd_strength/PasswordStrength.java (hunk 6 above)
Comment (siddharthtankariya) [reply passwd_strength/PasswordStrength.java]: Done

**Comment by t2013anurag** [comment]
@siddharthtankariya - delete the password_stength file. Only keep the file inside password_generator.

**Comment by siddharthtankariya** [comment]
yes i have deleted the folder.and moved only .java file into passowrd_generator folder

**Comment by siddharthtankariya** [comment]
@t2013anurag i have done it.

**Comment by t2013anurag** [comment]
https://github.com/hacktoberfest17/programming/pull/455/files

 shows me two files, can you check again :)

**Comment by siddharthtankariya** [comment]
@t2013anurag done.

**Comment by t2013anurag** [comment]
@siddharthtankariya - LGTM, Merging.

**Comment by siddharthtankariya** [comment]
@t2013anurag Thanks

**Pull request merged**